submissions/UA-girl/task_friends_app/js/app.js

```diff
@@ -0,0 +1,300 @@
+            this.restoreFromFilter();
+        } else {
+            this.filterStates.age = +maxAge;
+            this.cards = this.cards.filter(person => person.age <= +maxAge);
+        }
+    }
+
+    sortCards(value, filterName) {
+        this.resetSortingState();
+        switch (filterName) {
+            case 'age':
+                this.cards.sort((person1, person2) => person1.compareAge(person2));
+                break;
+            case 'name':
+                this.cards.sort((person1, person2) => person1.compareName(person2));
+                break;
+            case 'registration':
+                this.cards.sort((person1, person2) => person1.compareDateRegistration(person2));
+        }
+        this.sortingStates[filterName] = value;
+        if (value === 'down') {
+            this.cards.reverse()
+        }
+    }
+
+    nameFilter(name) {
+        name = name.trim().toLowerCase();
+        this.restoreFromFilter();
+        this.cards = this.cards.filter(person => person.name.first.toLowerCase()
+            .startsWith(name) || person.name.last.toLowerCase().startsWith(name));
+    }
+
+    disableEnablePageToggler() {
+        const [backPage, forwardPage] = document.querySelectorAll('.button-page');
+        if ((this.currentPage + 1) * CARDS_PER_PAGE >= this.cards.length) {
+            forwardPage.classList.add('disabled');
+            forwardPage.setAttribute('disabled', true);
+        } else {
+            forwardPage.removeAttribute('disabled');
+            forwardPage.classList.remove('disabled');
+        }
+        if (this.currentPage === 0) {
+            backPage.classList.add('disabled');
+            backPage.setAttribute('disabled', true);
+        } else {
+            backPage.classList.remove('disabled');
+            backPage.removeAttribute('disabled');
+        }
+        const span = document.querySelector('.page');
+        span.textContent = this.currentPage + 1;
+    }
+
+    resetApp() {
+        this.cards = cards.slice();
+        this.resetFilterState(true);
+        this.resetSortingState();
+        document.querySelector('.label-range').innerHTML = `Max age: ${MAX_AGE}`;
+        document.querySelector('.range').value = MAX_AGE;
+        document.querySelector('.search').value = '';
+        document.querySelectorAll('.radio').forEach(input => input.checked = false)
+    }
+
+    displayPeople() {
+        this.container.innerHTML = '';
+        let cardsRender;
+        if (this.cards.length > CARDS_PER_PAGE) {
+            cardsRender =
+                this.cards.slice(CARDS_PER_PAGE * this.currentPage, CARDS_PER_PAGE * (this.currentPage + 1));
+        } else {
+            cardsRender = this.cards.slice();
+        }
+        const fragment = document.createDocumentFragment();
+        cardsRender.forEach(person => fragment.append(person.card));
+        this.container.appendChild(fragment);
+        this.disableEnablePageToggler();
+    }
+
+    toggleFilters() {
```

Comment:
I'd recommend renaming this function. Maybe `setupFilters` or sth like this.

Reply:
O, I have a lot of problems with naming! ))) Thank you, I'll try to improve this )))

Reply:
Yeah, it's ok. I have these problems as well))